product_weight/models/product.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2024 Antoni Marroig(APSL-Nagarro)<amarroig@apsl.net>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    total_weight = fields.Float(
+        "Total Weight(kg)", compute="_compute_total_weight", store=True
+    )
+
+    @api.depends("qty_available")
+    def _compute_total_weight(self):
+        for product in self:
+            product.total_weight = product.qty_available * product.weight
+
+
+class Product(models.Model):
+    _inherit = "product.product"
+
+    total_weight = fields.Float(
+        "Total Weight(kg)", compute="_compute_total_weight", store=True
```

Comment:
this field can not be stored, as it depends on a non stored field. (qty_available) (same for field on product.template).